### PR TITLE
[docs] Update readme and documentation with captioning details

### PR DESCRIPTION
**Pull request by vedanuj**
Updating readme to add image captioning model. For issue #79

Added Colab notebook demo link for Captioning. [link](https://colab.research.google.com/drive/1vzrxDYB0vxtuUy8KCaGxm--nDCJvyBSg)

### CODE REVIEW

**Comment by apsdehal** [comment]
We should also update the file paths while we or int in this one.

File: README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Learn more about Pythia [here](https://learnpythia.readthedocs.io/en/latest/).
 
 ## Demo
 
-Try the demo for Pythia model on [Colab](https://colab.research.google.com/drive/1Z9fsh10rFtgWe4uy8nvU4mQmqdokdIRR).
+Try the demo for Pythia model on [Colab](https://colab.research.google.com/drive/1Z9fsh10rFtgWe4uy8nvU4mQmqdokdIRR). Try the demo of BUTD captioning model here on [Colab](https://colab.research.google.com/drive/1vzrxDYB0vxtuUy8KCaGxm--nDCJvyBSg).
```
Comment (apsdehal) [inline README.md]:
```suggestion
1. [Pythia VQA](https://colab.research.google.com/drive/1Z9fsh10rFtgWe4uy8nvU4mQmqdokdIRR). 
2. [BUTD Captioning](https://colab.research.google.com/drive/1vzrxDYB0vxtuUy8KCaGxm--nDCJvyBSg).
```

File: README.md [hunk 2]
```diff
@@ -107,13 +108,14 @@ supported by the models in Pythia's model zoo.
 | Pythia | pythia    | vqa2, vizwiz, textvqa | [vqa2 train+val](https://dl.fbaipublicfiles.com/pythia/pretrained_models/vqa2/pythia_train_val.pth), [vqa2 train only](https://dl.fbaipublicfiles.com/pythia/pretrained_models/vqa2/pythia.pth), [vizwiz](https://dl.fbaipublicfiles.com/pythia/pretrained_models/vizwiz/pythia_pretrained_vqa2.pth)  | VizWiz model has been pretrained on VQAv2 and transferred |
 | LoRRA  | lorra     | vqa2, vizwiz, textvqa       | [textvqa](https://dl.fbaipublicfiles.com/pythia/pretrained_models/textvqa/lorra_best.pth)      |                               |
 | BAN    | ban       | vqa2, vizwiz, textvqa | Coming soon!      | Support is preliminary and haven't been tested thoroughly. |
+| BUTD    | butd       | coco | [butd](https://dl.fbaipublicfiles.com/pythia/pretrained_models/coco_captions/butd.pth)    | For training use the config `butd.yml`. Training uses greedy decoding for validation. Currently we do not have support to train the model using beam search decoding validation. We will add that support soon. For inference only use `butd_beam_search.yml` config that supports beam search decoding. |
```
Comment (apsdehal) [inline README.md]: Too much information here. We should create a separate note somewhere.

File: README.md [hunk 3]
```diff
@@ -142,6 +144,7 @@ The table below shows inference metrics for various pretrained models:
 | Pythia | vqa2 (train)     | test-dev accuracy - 66.70%  |  |
 | Pythia | vizwiz (train)     | test-dev accuracy - 54.22%  |    Pretrained on VQA2 and transferred to VizWiz                           |
 | LoRRA  | textvqa (train)  | val accuracy - 27.4%       |                               |
+| BUTD  | coco  (karpathy train)  | BLEU 1 - 76.02, BLEU 4 - 35.42 , METEOR - 27.39, ROUGE_L - 56.17, CIDEr - 112.03 , SPICE -  20.33    |   The performance reported here is using beam search decoding (beam size of 5) inference on the Karpathy test split (using `butd_beam_search.yml` config).                           |
```
Comment (apsdehal) [inline README.md]:
```suggestion
| BUTD  | coco  (karpathy train)  | BLEU 1 - 76.02, BLEU 4 - 35.42 , METEOR - 27.39, ROUGE_L - 56.17, CIDEr - 112.03 , SPICE -  20.33    |   Beam Search 5, Karpathy test split |
```

**Comment by vedanuj** [comment]
Updated documentation and tested locally that the `rst` tables are built properly

**Pull request merged**